event-system/tests/subscribeSpec.js

```diff
@@ -1,42 +1,127 @@
+import _ from 'lodash';
 import test from 'blue-tape';
+import sinon from 'sinon';
+import express from 'express';
+import bodyparser from 'body-parser';
+import EventSource from 'eventsource';
 import request from 'supertest-as-promised';
-import {publicApi, privateApi} from '../src/index.js';
-import authorization from './jwtHelpers.js';
+import redis from './redis.js';
+import {fakeauth, profile, token, authorization} from './auth.js';
+import normalize from '../src/normalize.js';
+import subscribe from '../src/subscribe.js';
 
-const before = test;
-const after = test;
+const beforeEach = (dbName) => {
+  subscribe.__Rewire__('redis', {createClient: redis.createClient.bind(null, dbName)});
 
-before('before', (assert) => {
-  return new Promise((resolve) => {
-    resolve();
+  let app = express();
+  app.use(bodyparser.json());
+  app.use(fakeauth(0));
+  app.use(subscribe);
+  return app.listen(9000);
+};
+
+const url = (path) => {
+  return 'http://127.0.0.1:9000' + path;
+};
+
+test('Check authorization normalization for SSE requests', (t) => {
+  let app = express();
+  app.use(normalize);
+  app.get('/subscribe/:namespace/:room', (req, res) => {
+    t.equals(req.get('Authorization'), authorization(0), 'Request was normalized');
+    res.send();
   });
-});
 
-test('GET /subscriptions should respond with 200', (assert) => {
-  return request(publicApi)
-    .get('/subscriptions/namespace/room')
-    .set('Authorization', authorization[0])
+  return request(app)
+    .get('/subscribe/namespace/room/' + token(0))
     .expect(200);
 });
 
-test('DELETE /subscriptions should respond with 200', (assert) => {
-  return request(publicApi)
-    .delete('/subscriptions/namespace/room')
-    .set('Authorization', authorization[0])
-    .expect(200);
+test('GET /subscribe/:namespace/:room should receive events published to namespace:room', (t) => {
+  return new Promise((resolve) => {
+    let app = beforeEach('subscribe');
+    let client = redis.createClient('subscribe');
+    let spy = sinon.spy();
+
+    let es = new EventSource(url('/subscribe/namespace/room'));
```

Comment:
The client has to pass jwt as an url parameter. It's injected into the Authorization header before reaching the express-jwt middleware. The client can also register to the message event, which is emitted on every event. But essentially yes.